core/src/main/java/org/infinispan/topology/LocalTopologyManagerImpl.java

```diff
@@ -130,15 +130,17 @@ public CacheTopology join(String cacheName, CacheJoinInfo joinInfo, CacheTopolog
       // For Total Order caches, we must not move the topology updates to another thread
       ExecutorService topologyUpdatesExecutor = joinInfo.isTotalOrder() ? withinThreadExecutor : asyncTransportExecutor;
       LocalCacheStatus cacheStatus = new LocalCacheStatus(cacheName, joinInfo, stm, phm, topologyUpdatesExecutor);
-      runningCaches.put(cacheName, cacheStatus);
 
-      long timeout = joinInfo.getTimeout();
-      long endTime = timeService.expectedEndTime(timeout, TimeUnit.MILLISECONDS);
       // Pretend the join is using up a thread from the topology updates executor.
       // This ensures that the initial topology and the GET_CACHE_LISTENERS request will happen on this thread,
-      // and other topology updates are only handled after we call backgroundTaskFinished(null)
+      // and other topology updates are only handled after we complete joinFuture.
       CompletableFuture<Void> joinFuture = new CompletableFuture<>();
       cacheStatus.getTopologyUpdatesExecutor().executeAsync(() -> joinFuture);
+
```

Comment:
I don't see how initializing the  ```runningCaches``` before or after the sending the ```joinFuture``` to the executor could affect anything... 
My point is: I'm not seeing any difference in the logic executed. So, what side effect am I missing?